### ISSUE TITLE
Align text from wav2vec2

### ISSUE THREAD

**Issue by doublex** [lumaku/ctc-segmentation#11]
How to use ctc-segmentation with wav2vec2? 
The down below code works - but is not properly aligned.

This wav: [meisterfloh.zip](https://github.com/lumaku/ctc-segmentation/files/6564953/meisterfloh.zip)
This code:
````python
import torch, transformers, ctc_segmentation
import soundfile

# wav2vec2
model_file = 'facebook/wav2vec2-large-xlsr-53-german'
vocab_dict = {"<pad>": 0, "<s>": 1, "</s>": 2, "<unk>": 3, "|": 4, "E": 5, "N": 6, "I": 7, "S": 8, "R": 9, "T": 10, "A": 11, "H": 12, "D": 13, "U": 14, "L": 15, "C": 16, "G": 17, "M": 18, "O": 19, "B": 20, "W": 21, "F": 22, "K": 23, "Z": 24, "V": 25, "Ü": 26, "P": 27, "Ä": 28, "Ö": 29, "J": 30, "Y": 31, "'": 32, "X": 33, "Q": 34, "-": 35}

processor = transformers.Wav2Vec2Processor.from_pretrained( model_file )
model = transformers.Wav2Vec2ForCTC.from_pretrained( model_file )

speech_array, sampling_rate = soundfile.read( 'meisterfloh.wav' )
assert sampling_rate == 16000
features = processor(speech_array,sampling_rate=16000, return_tensors="pt")
input_values = features.input_values
attention_mask = features.attention_mask
with torch.no_grad():
    logits = model(input_values).logits
predicted_ids = torch.argmax(logits, dim=-1)
transcription = processor.batch_decode(predicted_ids)[0]
transcription = transcription.lower()

# ctc-segmentation
config = ctc_segmentation.CtcSegmentationParameters()
with torch.no_grad():
    softmax = torch.nn.Softmax(dim = -1)
    lpz = softmax(logits)[0].cpu().numpy()
char_list = [x.lower() for x in vocab_dict.keys()]
ground_truth_mat, utt_begin_indices = ctc_segmentation.prepare_text(config, transcription,char_list)
timings, char_probs, state_list = ctc_segmentation.ctc_segmentation(config, lpz, ground_truth_mat)
segments = ctc_segmentation.determine_utterance_segments(config, utt_begin_indices, char_probs, timings, transcription)

# dump
for word, segment in zip(transcription.split(' '), segments):
    print( word, segment )
````

**Comment by lumaku** [comment]
1. Set `index_duration` to obtain the correct timings:

```python
config.index_duration = 0.0200  # or: speech_array.shape[0] / lpz.shape[0] / sampling_rate
```

2. Change `Softmax` to `LogSoftmax`
```python
with torch.no_grad():
    softmax = torch.nn.LogSoftmax(dim = -1)
    lpz = softmax(logits)[0].cpu().numpy()
```

(I was surprised to see that the timings were also mostly correct with the regular softmax. Nevertheless, the algorithm was written for log probabilities for faster / more accurate computation.)

3. Also, the transcription must be splitted into the same format (list) at the alignment as the segments: 

```python
transcription = processor.batch_decode(predicted_ids)[0]
transcription = transcription.split()
# ...
for word, segment in zip(transcription, segments):  # note: removed split()
    print( f"{segment[0]:.2f} {segment[1]:.2f} {segment[2]:3.4f} {word}")
```
Then, the output:

```
0.11 0.49 -0.0095 worin
0.49 0.65 -0.5070 der
0.65 1.14 -0.3055 geneigte
1.14 1.55 -0.3945 leser
1.55 1.75 -0.7288 so
1.75 2.09 -0.1064 viel
2.09 2.29 -0.8035 aus
2.29 2.47 -0.8941 dem
2.47 2.79 -0.3612 leben
2.79 2.97 -0.9926 des
2.97 3.25 -0.5454 herrn
3.25 3.97 -0.2566 peregrinus
3.97 4.25 -0.4797 tyss
4.25 4.77 -0.2234 erfährt
4.84 5.47 -0.0002 als
5.47 5.67 -0.8421 ihm
5.67 5.85 -0.8207 zu
5.85 6.19 -0.3102 wissen
6.19 6.65 -0.2335 nötig
7.04 7.60 -0.0005 die
7.60 8.63 -0.2550 weihnachtsbescherung
8.63 8.81 -0.4637 bei
8.81 8.95 -0.5152 dem
```

**Comment by doublex** [comment]
The ctc-segmentation is genius. Your patch fixed this issue.
Sorry for the inconvenience.
